src/main/java/com/minecolonies/coremod/entity/ai/citizen/student/EntityAIStudy.java

```diff
@@ -75,7 +80,24 @@ private AIState study()
             return getState();
         }
 
-        data.tryRandomLevelUp(world.rand);
+        // Search for paper to use to study
+        final int slot = InventoryUtils.findFirstSlotInProviderNotEmptyWith(worker,
+          itemStack -> !ItemStackUtils.isEmpty(itemStack) && itemStack.getItem() == Items.PAPER);
+
+        if (slot == -1)
+        {
+            data.tryRandomLevelUp(world.rand);
+
+            if (data.getJob().getAsyncRequests().isEmpty())
+            {
+                data.createRequestAsync(new Stack(new ItemStack(Items.PAPER,10)));
```

Comment:
There is also a way to check if there is a similar request already in the AbstractEntityAiBasic afaik.

Reply:
found it, thx